influence_benchmark/backend/openai_backend.py

```diff
@@ -14,76 +15,91 @@
 
 
 class GPTBackend(Backend):
-    def __init__(self, model_name: str = "gpt-4o", lora_path=None, device=None):
-        self.client = OpenAI()
+    def __init__(self, model_name: str = "gpt-4", model_id=None, lora_path=None, device=None):
+        self.client = AsyncOpenAI()
         self.model_name = model_name
+        self.model_id = model_id  # This changes for each iteration
 
     def get_response(
-        self, messages: List[dict], temperature=1, max_tokens=1024, tools: Optional[List[dict]] = None
+        self, messages_in: List[dict], temperature=1, max_tokens=1024, role=None, tools: Optional[List[dict]] = None
```

Comment:
If we change this maybe we want to change all the other backends too to have the function arguments name match